grow/pods/documents.py

```diff
@@ -264,6 +264,13 @@ def locales(self):
             if codes is None:
                 return []
             return locales.Locale.parse_codes(codes)
+        # If `$localization` is missing, use locales inferred from document
```

Comment:
Jumping back into this. The test case in question is the one covered by this testdata:

https://github.com/grow/grow/blob/master/grow/testing/testdata/pod/content/pages/about.yaml

```
---
$title@: About 
$titles:
  nav@: About us
$order: 1.1
foo: bar
qaz: qux
---
$locale: de
$title@: AboutDE
foo: baz
```

With the changes we're making in this PR, this document would no longer generate locales other than `de`, which I'm not sure is right. The current behavior is that the document inherits the blueprint's localization path format, and the collection inherits the podspec's locales.

#157 is slightly different from this scenario I think. In #157 the bug is the text in the localized document part is extracted for all locales (definitely a bug), in this scenario the issue at hand is should specifying a single locale override also override the inheritance of locales from document to collection to podspec?

That's kind of why I thought appending might be the right approach, rather than replacing. (Or, in theory we could analyze the locales in parts and do something like, "if not in collection locales, replace; if in collection locales, override". But that might be too magical.